library/nist_kw.c

```diff
@@ -0,0 +1,650 @@
+/**
+ * \file kw.c
+ *
+ * \brief This file provides an API key wrapping(KW)
+ *        and key wrapping with padding(KWP) as defined in NIST SP800-38F
+ *        https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-38F.pdf
```

Comment:
We don't usually put Doxygen comments in our C files (as we don't run Doxygen on our C files). Consider revising to match existing commenting style. Also consider removing from here what is not implementation-specific.

It looks to me like all this information can go into the header file (if it isn't already there) and nothing here is implementation specific.